### PR TITLE
Disallow self as target chain

### LINKED ISSUE #496: Implement check in contracts for same source and targets chain ids
The source and target chain ids should not be the same.

### PR DESCRIPTION
**Pull request by palango**
Resolves #496 

We don't need an explicit check, because we have a check here now: https://github.com/beamer-bridge/beamer/blob/082a23ad9a956f61e0861e132f6b26603a8f42ba/contracts/contracts/RequestManager.sol#L158

I also appended a little commit that re-orders the requires, so that we burn less gas if a require should be hit.

### CODE REVIEW

File: scripts/deployment/main.py [hunk 1]
```diff
@@ -214,7 +215,11 @@ def main(
         chain_id = web3_l2.eth.chain_id
         assert chain_id == l2_config["chain_id"]
 
-        supported_target_chains = {l2["chain_id"]: l2["finalization_time"] for l2 in config["L2"]}
+        supported_target_chains = {
+            l2["chain_id"]: l2["finalization_time"]
+            for l2 in config["L2"]
+            if l2["chain_id"] != chain_id or chain_id == GANACHE_CHAIN_ID
+        }
```
Comment (istankovic) [inline scripts/deployment/main.py:222]: How has it been working so far, without explicitly mentioning ganache?

Comment (istankovic) [reply scripts/deployment/main.py:222]: I've been using
```
--config-file scripts/deployment/ganache-local.json --test-messenger
```
as options to the deployment script for some time now and it worked for me.

Comment (palango) [reply scripts/deployment/main.py:222]: You removed the if clause in your PR: https://github.com/beamer-bridge/beamer/pull/488/files#diff-c6d950f3bb92a40d8b76462f0c0f34448141fee95c49373f1e35d96960a48828R217

So I bring it back and make sure that it still works on local chains.

Comment (istankovic) [reply scripts/deployment/main.py:222]: Ah, I see, you needed to bring it in due to it being the actual topic of the PR -- and then you need to allow for ganache again. Understood. :+1:

**Review by istankovic — APPROVED** [review]

**Pull request merged**